cli/src/update/unix.rs

```diff
@@ -69,6 +70,17 @@ async fn http_get(url: &str) -> anyhow::Result<reqwest::Response> {
     Ok(response)
 }
 
+/// Generic function for fetching JSON structs via HTTP GET.
+async fn http_get_json<T: DeserializeOwned>(url: &str) -> anyhow::Result<T> {
+    let response = http_get(url).await?;
+
+    if let Err(error) = response.error_for_status_ref() {
+        Err(anyhow!(response.text().await?)).context(error)
+    } else {
+        Ok(response.json::<T>().await?)
```

Comment:
I assume the type inference engine can figure out the `::<T>` here. Or am I missing something?

```suggestion
        Ok(response.json().await?)
```